src/lib/MetadataLib.sol

```diff
@@ -81,88 +81,439 @@ library MetadataLib {
+            _getTextPath(token, "-100%"),
+            _getTextPath(token, "0%"),
+            _getTextPath(allocator, "50%"),
+            _getTextPath(allocator, "-50%"),
+            "</text>"
+        );
+    }
+
+    /**
+     * @notice Builds a textPath element that moves along the border.
+     * @param text The text to display.
+     * @param startOffset The starting offset of the text.
+     * @return The SVG animated text.
+     */
+    function _getTextPath(string memory text, string memory startOffset) internal pure returns (string memory) {
+        return string.concat(
+            '<textPath startOffset="',
+            startOffset,
+            '" fill="white" font-family="monospace" font-size="10px" xlink:href="#tp">',
+            text,
+            '<animate additive="sum" attributeName="startOffset" from="0%" to="100%" begin="0s" dur="30s" repeatCount="indefinite" />',
+            "</textPath>"
+        );
+    }
+
+    /**
+     * @notice Returns the SVG title section.
+     * @param lock The lock.
+     * @return The SVG title section.
+     */
+    function _getSvgTitleSection(Lock memory lock) internal view returns (string memory) {
+        (,, string memory tokenSymbol,) = _getTokenDetails(lock);
+        string memory scope = lock.scope.toString();
+        string memory lockId = LibString.toHexString(toId(lock));
+        return string.concat(
+            "<g id=\"title\">",
+            '<text y="60px" x="32px" fill="white" font-family="monospace" font-weight="100" font-size="32px" filter="url(#ts)">',
+            "Compact ",
+            tokenSymbol,
+            "</text>",
+            '<text y="90px" x="32px" fill="rgba(255,255,255,0.6)" font-family="monospace" font-weight="50" font-size="22px" filter="url(#ts)">',
+            scope,
+            " Resource Lock",
+            "</text>",
+            '<text y="110px" x="32px" fill="rgba(255,255,255,0.6)" font-family="monospace" font-weight="100" font-size="10px" filter="url(#ts)">',
+            "ID: ",
+            lockId,
+            "</text>",
+            "</g>"
+        );
+    }
+
+    /**
+     * @notice Returns the SVG details section
+     * @dev This section contains the details of the lock, including the token, allocator, and reset period.
+     * @param lock The lock.
+     * @return The SVG details section.
+     */
+    function _getSvgDetailsSection(Lock memory lock) internal view returns (string memory) {
+        (, string memory tokenName, string memory tokenSymbol,) = _getTokenDetails(lock);
+        string memory allocatorName = _tryReadAllocatorName(lock.allocator);
+        string memory resetPeriod = lock.resetPeriod.toString();
+        string memory scope = lock.scope.toString();
+        // Handshake Icon
+        string memory iconSvg = string.concat(
+            '<g style="transform:translate(420px, 50px)">',
+            '<text x="20px" y="28px" text-anchor="middle" font-size="64px" opacity="0.1">',
```

Comment:
How come this has opacity?

Reply:
The handshake icon has 10% opacity so it's unobtrusive and doesn't draw focus immediately. I'm no designer but
<img width="502" alt="image" src="https://github.com/user-attachments/assets/aaeea045-f5df-4f7b-bc68-336031c310b0" />
reads and looks much better than
<img width="501" alt="image" src="https://github.com/user-attachments/assets/8ef3f615-22ca-40c6-b297-70f23ee6c57b" />

Reply:
I like the version without opacity haha

maybe we split the difference?

Reply:
how about 33%?

Reply:
better!